flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/controller/RollbackTest.java

```diff
@@ -179,6 +181,17 @@ public void testRollbackFailureWithLastState(FlinkVersion flinkVersion) throws E
                     assertEquals("RUNNING", dep.getStatus().getJobStatus().getState());
                     assertEquals(1, flinkService.listJobs().size());
 
+                    // Trigger deployment recovery
+                    flinkService.clear();
+                    flinkService.setPortReady(false);
+
+                    testController.reconcile(dep, context);
+                    flinkService.setPortReady(true);
+                    testController.reconcile(dep, context);
+                    var jobs = flinkService.listJobs();
+                    // Make sure deployment was recovered with correct spec/config
+                    assertTrue(jobs.get(jobs.size() - 1).f2.containsKey("t"));
```

Comment:
Not really, this change make it so that you pick up `kubernetes.operator.*` configs from the latest spec so even if you roll back you still pick the same.